src/azure-cli/azure/cli/command_modules/network/tests/latest/test_network_commands.py

```diff
@@ -2192,6 +2192,133 @@ def test_network_ag_routing_rule(self, resource_group):
+            'port': 8080,
+        })
+
+        self.cmd('network public-ip create '
+                 '--resource-group {rg} '
+                 '--name {public-ip-name} '
+                 '--sku Standard')
+
+        self.cmd('network vnet create '
+                 '--resource-group {rg} '
+                 '--name {vnet-name} '
+                 '--address-prefix {vnet-address}')
+
+        self.cmd('network vnet subnet create '
+                 '--resource-group {rg} '
+                 '--name {subnet-name} '
+                 '--vnet-name {vnet-name} '
+                 '--address-prefix {subnet-address} '
+                 '--default-outbound false')
+
+        self.cmd('network application-gateway create '
+                 '--resource-group {rg} '
+                 '--name {gateway-name} '
+                 '--sku Standard_v2 '
+                 '--public-ip-address {public-ip-name} '
+                 '--vnet-name {vnet-name} '
+                 '--subnet {subnet-name} '
+                 '--priority 1001')
+
+        self.cmd('network application-gateway settings create '
+                 '--resource-group {rg} '
+                 '--name {setting-name} '
+                 '--gateway-name {gateway-name} '
+                 '--port {port}')
+
+        self.cmd('network application-gateway settings show '
+                 '--resource-group {rg} '
+                 '--name {setting-name} '
+                 '--gateway-name {gateway-name}',
+                 checks=(self.check('enableL4ClientIpPreservation', False)))
+
+        self.cmd('network application-gateway settings update '
+                 '--resource-group {rg} '
+                 '--name {setting-name} '
+                 '--gateway-name {gateway-name} '
+                 '--enable-l4-client-ip-preservation true',
+                 checks=(self.check('enableL4ClientIpPreservation', True)))
+
+        setting_list = self.cmd('network application-gateway settings list '
+                 '--resource-group {rg} '
+                 '--gateway-name {gateway-name}').get_output_in_json()
+
+        self.assertTrue(len(setting_list) > 0)
+
+    @ResourceGroupPreparer(name_prefix='cli_test_agp_new_prop')
+    def test_network_ags_enable_probe_proxy_protocol_header(self, resource_group):
+        self.kwargs.update({
+            'public-ip-name': self.create_random_name(prefix='public',length=15),
+            'vnet-name': self.create_random_name(prefix='vnet',length=15),
+            'vnet-address': '10.10.0.0/16',
+            'subnet-address': '10.10.0.0/24',
+            'subnet-name': self.create_random_name(prefix='subnet',length=15),
+            'gateway-name': self.create_random_name(prefix='gateway',length=15),
+            'probe-name': self.create_random_name(prefix='probe',length=15),
+        })
+
+        self.cmd('network public-ip create '
+                 '--resource-group {rg} '
+                 '--name {public-ip-name} '
+                 '--sku Standard')
```

Comment:
we'd better input `--ip-tags FirstPartyUsage=/NonProd` when creating a public ip resource.